lib/ex_oauth2_provider/redirect_uri.ex

```diff
@@ -26,19 +26,25 @@ defmodule ExOauth2Provider.RedirectURI do
 
   defp do_validate(_url, %{fragment: fragment}, _config) when not is_nil(fragment),
     do: {:error, "Redirect URI cannot contain fragments"}
-  defp do_validate(_url, %{scheme: schema, host: host}, _config) when is_nil(schema) or is_nil(host),
+  defp do_validate(_url, %{scheme: scheme, host: _host}, _config) when is_nil(scheme),
     do: {:error, "Redirect URI must be an absolute URI"}
-  defp do_validate(url, uri, config) do
-    if invalid_ssl_uri?(uri, config) do
+  defp do_validate(_url, %{scheme: "https", host: host}, _config) when is_nil(host) or host == "",
+    do: {:error, "Redirect URI must be an absolute URI"}
+  defp do_validate(_url, %{scheme: "http", host: host}, _config) when is_nil(host) or host == "",
```

Comment:
The `is_nil(host) or host == ""` is required due to this bug I found in elixir elixir-lang/elixir#9837